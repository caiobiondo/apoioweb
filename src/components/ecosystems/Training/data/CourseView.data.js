import gql from 'graphql-tag';
import { Origem } from 'config';

export const CourseViewQuery = gql`
  query CourseViewQuery(
    $sellerId: Int!
    $courseId: Int!
    $ciclo: Int
    $setor: Int
    $gerenciaMercado: Int
    $grupo: Int
    $papelDaConsultora: Int
    $canal: Int
    $gerenciaDeVendas: Int
    $regiao: Int
    $origem: String!
  ) {
    course(
      sellerId: $sellerId
      courseId: $courseId
      ciclo: $ciclo
      setor: $setor
      gerenciaMercado: $gerenciaMercado
      grupo: $grupo
      papelDaConsultora: $papelDaConsultora
      canal: $canal
      gerenciaDeVendas: $gerenciaDeVendas
      regiao: $regiao
      origem: $origem
    ) {
      id
      accessToken
      categoryTitle
      clientIdentifier
      clientSecrets
      courseContent {
        video
        html5
        web
        videoEmbedUrl
        scorm
      }
      dateUpload
      durationInSeconds
      description
      generalRating
      isfavorite
      myRating
      ratedByYou
      relatedCourses {
        id
        title
        thumbnail
        type
        url
      }
      status
      stoppedAt
      thumbnail
      title
      type
      views
    }
  }
`;

export const CourseViewQueryOptions = {
  options(props) {
    return {
      variables: {
        sellerId: props.user.codigo,
        courseId: props.courseId,
        ciclo:
          props.user.estrutura.ciclo &&
          props.user.estrutura.ciclo[0] &&
          props.user.estrutura.ciclo[0].numero,
        setor: props.user.estrutura.codigoTipo > 3 ? props.user.estrutura.setor.codigo : 0,
        gerenciaMercado: props.user.estrutura.gerenciaMercado.codigo,
        grupo: props.user.estrutura.codigoTipo > 4 ? props.user.estrutura.codigo : 0,
        papelDaConsultora: props.user.cdPapelAtivo,
        canal: props.user.cdCanalCaptacao,
        gerenciaDeVendas:
          props.user.estrutura.codigoTipo > 2 ? props.user.estrutura.gerenciaVenda.codigo : 0,
        regiao:
          props.user.estrutura.codigoTipo > 1 ? props.user.estrutura.regiaoEstrategica.codigo : 0,
        origem: Origem,
      },
      fetchPolicy: 'cache-and-network',
      //fetchPolicy: 'no-cache',
    };
  },

  props({ data }) {
    return {
      data,
      loading: data.loading,
      course: data.course,
      refetch: data.refetch,
    };
  },
};
