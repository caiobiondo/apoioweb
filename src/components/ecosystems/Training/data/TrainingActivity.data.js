import gql from 'graphql-tag';
import { Origem } from 'config';

export const ActivityViewQuery = gql`
  query ActivityViewQuery(
    $sellerId: Int!
    $activityId: Int!
    $ciclo: Int
    $setor: Int
    $gerenciaMercado: Int
    $grupo: Int
    $papelDaConsultora: Int
    $canal: Int
    $gerenciaDeVendas: Int
    $regiao: Int
    $origem: String!
  ) {
    activity(
      sellerId: $sellerId
      activityId: $activityId
      ciclo: $ciclo
      setor: $setor
      gerenciaMercado: $gerenciaMercado
      grupo: $grupo
      papelDaConsultora: $papelDaConsultora
      canal: $canal
      gerenciaDeVendas: $gerenciaDeVendas
      regiao: $regiao
      origem: $origem
    ) {
      status
      message
      totalOfElement
      results {
        id
        finished
        name
        type
        accessToken
        courseContent {
          video
          html5
          web
          videoEmbedUrl
          html5Embed
          scorm
        }
        assessment {
          id
          title
          questions {
            id
            description
            type
            alternatives {
              id
              description
            }
          }
        }
      }
    }
  }
`;

export const ActivityViewQueryOptions = {
  options(props) {
    return {
      variables: {
        sellerId: props.user.codigo,
        activityId: props.activityId,
        ciclo:
          props.user.estrutura.ciclo &&
          props.user.estrutura.ciclo[0] &&
          props.user.estrutura.ciclo[0].numero,
        setor: props.user.estrutura.codigoTipo > 3 ? props.user.estrutura.setor.codigo : 0,
        gerenciaMercado: props.user.estrutura.gerenciaMercado.codigo,
        grupo: props.user.estrutura.codigoTipo > 4 ? props.user.estrutura.codigo : 0,
        papelDaConsultora: props.user.cdPapelAtivo,
        canal: props.user.cdCanalCaptacao,
        gerenciaDeVendas:
          props.user.estrutura.codigoTipo > 2 ? props.user.estrutura.gerenciaVenda.codigo : 0,
        regiao:
          props.user.estrutura.codigoTipo > 1 ? props.user.estrutura.regiaoEstrategica.codigo : 0,
        origem: Origem,
      },
      fetchPolicy: 'network-only',
    };
  },

  props({ data }) {
    return {
      data,
      loading: data.loading,
      activity: data.activity,
      refetch: data.refetch,
    };
  },
};
