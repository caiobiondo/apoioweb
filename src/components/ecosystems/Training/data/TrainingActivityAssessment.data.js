import gql from 'graphql-tag';
export const TrainingActivityUpdateMutation = gql`
  mutation AddActivityAnswers(
    $sellerId: Int!
    $activityId: Int!
    $input: AddActivityAnswersInput!
    $ciclo: Int!
    $roleId: Int
    $grupo: Int!
    $gerenciaDeVendas: Int!
    $regiao: Int!
    $setor: Int!
    $gerenciaMercado: Int!
    $papelDaConsultora: Int!
    $canal: Int!
    $origem: String!
  ) {
    addActivityAnswers(
      sellerId: $sellerId
      activityId: $activityId
      input: $input
      ciclo: $ciclo
      roleId: $roleId
      grupo: $grupo
      gerenciaDeVendas: $gerenciaDeVendas
      gerenciaMercado: $gerenciaMercado
      regiao: $regiao
      setor: $setor
      papelDaConsultora: $papelDaConsultora
      canal: $canal
      origem: $origem
    ) {
      action
      stoppedAt
    }
  }
`;
