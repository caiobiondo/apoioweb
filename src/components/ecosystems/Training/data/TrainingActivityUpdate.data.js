import gql from 'graphql-tag';

export const TrainingActivityUpdateMutation = gql`
  mutation UpdateActivity(
    $sellerId: Int!
    $activityId: Int!
    $input: UpdateActivityInput!
    $ciclo: Int!
    $roleId: Int
    $grupo: Int!
    $gerenciaDeVendas: Int!
    $regiao: Int!
    $setor: Int!
    $gerenciaMercado: Int!
    $papelDaConsultora: Int!
    $canal: Int!
    $origem: String!
  ) {
    updateActivity(
      sellerId: $sellerId
      activityId: $activityId
      input: $input
      ciclo: $ciclo
      roleId: $roleId
      grupo: $grupo
      gerenciaDeVendas: $gerenciaDeVendas
      gerenciaMercado: $gerenciaMercado
      regiao: $regiao
      setor: $setor
      papelDaConsultora: $papelDaConsultora
      canal: $canal
      origem: $origem
    ) {
      status
      message
    }
  }
`;
