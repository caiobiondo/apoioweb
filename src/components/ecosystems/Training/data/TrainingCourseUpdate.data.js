import gql from 'graphql-tag';

export const TrainingCourseUpdateMutation = gql`
  mutation UpdateCourse(
    $sellerId: Int!
    $courseId: Int!
    $input: UpdateCourseInput!
    $ciclo: Int!
    $roleId: Int
    $grupo: Int!
    $gerenciaDeVendas: Int!
    $regiao: Int!
    $setor: Int!
    $gerenciaMercado: Int!
    $papelDaConsultora: Int!
    $canal: Int!
    $appVersion: String!
    $origem: String!
  ) {
    updateCourse(
      sellerId: $sellerId
      courseId: $courseId
      input: $input
      ciclo: $ciclo
      roleId: $roleId
      grupo: $grupo
      gerenciaDeVendas: $gerenciaDeVendas
      gerenciaMercado: $gerenciaMercado
      regiao: $regiao
      setor: $setor
      papelDaConsultora: $papelDaConsultora
      canal: $canal
      appVersion: $appVersion
      origem: $origem
    ) {
      status
      message
    }
  }
`;
