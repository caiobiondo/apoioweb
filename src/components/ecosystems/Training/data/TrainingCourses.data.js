import gql from 'graphql-tag';
import { Origem } from 'config';

const ITEMS_PER_PAGE = 20;

export const TrainingCoursesQuery = gql`
  query TrainingCoursesQuery(
    $sellerId: Int!
    $offset: Int!
    $limit: Int!
    $status: String
    $favorite: Boolean
    $filter: String
    $ciclo: Int
    $setor: Int
    $gerenciaMercado: Int
    $grupo: Int
    $papelDaConsultora: Int
    $canal: Int
    $gerenciaDeVendas: Int
    $regiao: Int
    $origem: String!
  ) {
    courses(
      sellerId: $sellerId
      offset: $offset
      limit: $limit
      status: $status
      favorite: $favorite
      filter: $filter
      ciclo: $ciclo
      setor: $setor
      gerenciaMercado: $gerenciaMercado
      grupo: $grupo
      papelDaConsultora: $papelDaConsultora
      canal: $canal
      gerenciaDeVendas: $gerenciaDeVendas
      regiao: $regiao
      origem: $origem
    ) {
      hasNextPage
      items {
        id
        accessToken
        categoryTitle
        clientIdentifier
        clientSecrets
        courseContent {
          video
          html5
          web
        }
        dateUpload
        durationInSeconds
        isfavorite
        ratedByYou
        status
        stoppedAt
        thumbnail
        title
        type
        views
      }
    }
  }
`;

export const updateQuery = (previousResult, { fetchMoreResult }) => {
  if (!fetchMoreResult) {
    return previousResult;
  }

  return Object.assign({}, previousResult, {
    courses: {
      __typename: previousResult.courses.__typename,
      hasNextPage: fetchMoreResult.courses.hasNextPage,
      items: [...previousResult.courses.items, ...fetchMoreResult.courses.items],
    },
  });
};

export const TrainingCoursesQueryOptions = {
  options(props) {
    return {
      variables: {
        sellerId: props.user.codigo,
        limit: ITEMS_PER_PAGE,
        offset: 0,
        status: props.status,
        favorite: props.favorite,
        filter: props.courseFilter,
        ciclo:
          props.user.estrutura.ciclo &&
          props.user.estrutura.ciclo[0] &&
          props.user.estrutura.ciclo[0].numero,
        setor: props.user.estrutura.codigoTipo > 3 ? props.user.estrutura.setor.codigo : 0,
        gerenciaMercado: props.user.estrutura.gerenciaMercado.codigo,
        grupo: props.user.estrutura.codigoTipo > 4 ? props.user.estrutura.codigo : 0,
        papelDaConsultora: props.user.cdPapelAtivo,
        canal: props.user.cdCanalCaptacao,
        gerenciaDeVendas:
          props.user.estrutura.codigoTipo > 2 ? props.user.estrutura.gerenciaVenda.codigo : 0,
        regiao:
          props.user.estrutura.codigoTipo > 1 ? props.user.estrutura.regiaoEstrategica.codigo : 0,
        origem: Origem,
      },
      fetchPolicy: 'cache-first',
    };
  },
  props({ data }) {
    const { refetch, loading } = data;
    const courses = data.courses && data.courses.items;
    const hasNextPage = (data.courses && data.courses.hasNextPage) || false;

    return {
      data,
      refetch,
      loading,
      courses,
      hasNextPage,
      fetchMore() {
        if (data.loading) {
          return;
        }

        const offset = courses ? courses.length : 0;
        const variables = { offset };

        return data.fetchMore({
          variables,
          updateQuery,
        });
      },
    };
  },
};
