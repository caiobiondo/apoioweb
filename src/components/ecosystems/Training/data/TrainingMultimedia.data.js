import gql from 'graphql-tag';
import { Origem } from 'config';

const ITEMS_PER_PAGE = 20;

export const TrainingMultimediaQuery = gql`
  query TrainingMultimediaQuery(
    $sellerId: Int!
    $offset: Int!
    $limit: Int!
    $status: String
    $favorite: Boolean
    $filter: String
    $ciclo: Int
    $setor: Int
    $gerenciaMercado: Int
    $grupo: Int
    $papelDaConsultora: Int
    $canal: Int
    $gerenciaDeVendas: Int
    $regiao: Int
    $origem: String!
  ) {
    trainingMultimedias(
      sellerId: $sellerId
      offset: $offset
      limit: $limit
      status: $status
      favorite: $favorite
      filter: $filter
      ciclo: $ciclo
      setor: $setor
      gerenciaMercado: $gerenciaMercado
      grupo: $grupo
      papelDaConsultora: $papelDaConsultora
      canal: $canal
      gerenciaDeVendas: $gerenciaDeVendas
      regiao: $regiao
      origem: $origem
    ) {
      hasNextPage
      items {
        id
        title
        description
        type
        date_upload
        duration_in_seconds
        thumbnail
        categories
        content
      }
    }
  }
`;

export const updateQuery = (previousResult, { fetchMoreResult }) => {
  if (!fetchMoreResult) {
    return previousResult;
  }

  return Object.assign({}, previousResult, {
    multimedias: {
      __typename: previousResult.multimedias.__typename,
      hasNextPage: fetchMoreResult.multimedias.hasNextPage,
      items: [...previousResult.multimedias.items, ...fetchMoreResult.multimedias.items],
    },
  });
};

export const TrainingMultimediaQueryOptions = {
  options(props) {
    return {
      variables: {
        sellerId: props.user.codigo,
        limit: ITEMS_PER_PAGE,
        offset: 0,
        favorite: props.favorite,
        filter: props.courseFilter,
        ciclo:
          props.user.estrutura.ciclo &&
          props.user.estrutura.ciclo[0] &&
          props.user.estrutura.ciclo[0].numero,
        setor: props.user.estrutura.codigoTipo > 3 ? props.user.estrutura.setor.codigo : 0,
        gerenciaMercado: props.user.estrutura.gerenciaMercado.codigo,
        grupo: props.user.estrutura.codigoTipo > 4 ? props.user.estrutura.codigo : 0,
        papelDaConsultora: props.user.cdPapelAtivo,
        canal: props.user.cdCanalCaptacao,
        gerenciaDeVendas:
          props.user.estrutura.codigoTipo > 2 ? props.user.estrutura.gerenciaVenda.codigo : 0,
        regiao:
          props.user.estrutura.codigoTipo > 1 ? props.user.estrutura.regiaoEstrategica.codigo : 0,
        origem: Origem,
      },
      fetchPolicy: 'cache-first',
    };
  },
  props({ data }) {
    const { refetch, loading } = data;
    const multimedias = data.multimedias && data.multimedias.items;
    const hasNextPage = (data.multimedias && data.multimedias.hasNextPage) || false;

    return {
      data,
      refetch,
      loading,
      multimedias,
      hasNextPage,
      fetchMore() {
        if (data.loading) {
          return;
        }

        const offset = multimedias ? multimedias.length : 0;
        const variables = { offset };

        return data.fetchMore({
          variables,
          updateQuery,
        });
      },
    };
  },
};
