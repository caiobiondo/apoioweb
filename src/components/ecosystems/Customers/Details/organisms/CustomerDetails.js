import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { graphql } from 'react-apollo';
import { injectIntl, FormattedMessage } from 'react-intl';
import { CustomerDetailsQuery, CustomerDetailsQueryOptions } from './CustomerDetails.data';
import withAuthErrorHandler from 'hocs/withAuthErrorHandler/withAuthErrorHandler';
import withUserData from 'hocs/withUserData/withUserData';
import CustomerAvatar from 'components/atoms/CustomerAvatar';
import {
  Main,
  CustomerDetailsWrapper,
  CustomerDetailsData,
  CustomerDetailsSectionTitleWrapper,
  CustomerAvatarWrapper,
  CustomerAvatarStyle,
  CustomerDataWrapper,
  CustomerData,
  CustomerNameWrapper,
  CustomerName,
  CustomerEditWrapper,
  CustomerDataTitle,
  CustomerDatumTelephoneNumber,
  CustomerDatumTelephoneProvider,
  CustomerDatumTelephone,
  CustomerDatumHalfWrapper,
  CustomerDataTelephones,
  CustomerDataAddresses,
  CustomerDatumAddress,
  CustomerDatumNotes,
} from './CustomerDetails.styles';
import SectionTitle from 'components/molecules/SectionTitle/SectionTitle';
import CustomerDatum from 'components/ecosystems/Customers/Details/molecules/CustomerDatum/CustomerDatum';
import { CustomerDatumValue } from 'components/ecosystems/Customers/Details/molecules/CustomerDatum/CustomerDatum.styles';

import { Paper, Loading, Icon } from 'natura-ui';

export class CustomerDetails extends Component {
  componentDidMount() {
    window.scrollTo(0, 0);
  }

  renderAddresses(addresses) {
    if (!addresses || !addresses.length) {
      return <CustomerDatumAddress>-</CustomerDatumAddress>;
    }

    return addresses.map((address, index) => {
      return this.renderAddress(address, index);
    });
  }

  renderAddress(address, index) {
    return (
      <CustomerDatumAddress key={index}>
        <CustomerDatumValue>{this.renderStreetName(address)}</CustomerDatumValue>
        <CustomerDatumValue>{this.renderNeighborhoodZipcode(address)}</CustomerDatumValue>
        <CustomerDatumValue>{this.renderCityState(address)}</CustomerDatumValue>
      </CustomerDatumAddress>
    );
  }

  renderPhone({ phone, provider }, index) {
    return (
      <CustomerDatumTelephone key={index}>
        <CustomerDatumTelephoneNumber>{phone || '-'}</CustomerDatumTelephoneNumber>
        <CustomerDatumTelephoneProvider>{provider || '-'}</CustomerDatumTelephoneProvider>
      </CustomerDatumTelephone>
    );
  }

  renderPhones(phones) {
    if (!phones || !phones.length || phones.length <= 1) {
      return <CustomerDatumTelephone>-</CustomerDatumTelephone>;
    }

    return phones.slice(1).map((phone, index) => {
      return this.renderPhone(phone, index);
    });
  }

  /* eslint-disable camelcase */
  renderStreetName({ street_name, street_number }) {
    return `${street_name}, ${street_number}`;
  }
  /* eslint-enable camelcase */

  renderNeighborhoodZipcode({ neighborhood, zipcode }) {
    return `${neighborhood} - ${zipcode}`;
  }

  renderCityState({ city, state }) {
    return `${city} - ${state}`;
  }

  parseBirthday(date) {
    if (!date) {
      return null;
    }
    const dateParts = date.split('-');
    if (dateParts.length !== 3) {
      return null;
    }
    return `${dateParts[2]}/${dateParts[1]}/${dateParts[0]}`;
  }

  render() {
    const { data: { customer, loading } } = this.props;

    if (loading) {
      return <Loading background="transparent" />;
    }

    const gender = customer.gender ? <FormattedMessage id={`gender.${customer.gender}`} /> : '-';

    const email = customer.emails && customer.emails[0] && customer.emails[0].email;
    const phone = customer.phones && customer.phones[0] && customer.phones[0].phone;
    const phoneProvider = customer.phones && customer.phones[0] && customer.phones[0].provider;
    const profileEditUrl = `/edit-customer/${customer.id}`;
    const name = customer.name || customer.nickname;
    const birthday = this.parseBirthday(customer.birthday);

    return (
      <Main>
        <Paper style={CustomerDetailsWrapper}>
          <CustomerDetailsSectionTitleWrapper>
            <SectionTitle iconName="ico_info" value="customerProfile" />
          </CustomerDetailsSectionTitleWrapper>
          <CustomerDetailsData>
            <CustomerAvatarWrapper>
              <CustomerAvatar name={name} size={115} style={CustomerAvatarStyle} />
            </CustomerAvatarWrapper>
            <CustomerDataWrapper>
              <CustomerData primary>
                <CustomerNameWrapper>
                  <CustomerName>{name}</CustomerName>
                  <CustomerEditWrapper>
                    <Link to={profileEditUrl}>
                      <Icon file="ico_pencil" />
                    </Link>
                  </CustomerEditWrapper>
                </CustomerNameWrapper>
                <CustomerDatumValue>{email || '-'}</CustomerDatumValue>
                <CustomerDatumValue>{phone || '-'}</CustomerDatumValue>
                <CustomerDatumValue>{phoneProvider || '-'}</CustomerDatumValue>
              </CustomerData>
              <CustomerData secondary>
                <CustomerDatum label="customerFullName" value={customer.name || '-'} />
                <CustomerDatumHalfWrapper>
                  <CustomerDatum label="customerBirthdate" value={birthday || '-'} />
                  <CustomerDatum label="customerGender" value={gender} />
                </CustomerDatumHalfWrapper>
              </CustomerData>
              <CustomerData secondary>
                <CustomerDataTelephones>
                  <CustomerDataTitle>
                    <FormattedMessage id="customerOtherTelephoneNumbers" />
                  </CustomerDataTitle>
                  {this.renderPhones(customer.phones)}
                </CustomerDataTelephones>
                <CustomerDataAddresses>
                  <CustomerDataTitle>
                    <FormattedMessage id="customerAddress" />
                  </CustomerDataTitle>
                  {this.renderAddresses(customer.addresses)}
                </CustomerDataAddresses>
              </CustomerData>
              <CustomerData secondary>
                <CustomerDatumNotes>
                  <CustomerDataTitle>
                    <FormattedMessage id="customerNotes" />
                  </CustomerDataTitle>
                  <CustomerDatumValue>{customer.comment || '-'}</CustomerDatumValue>
                </CustomerDatumNotes>
              </CustomerData>
            </CustomerDataWrapper>
          </CustomerDetailsData>
        </Paper>
      </Main>
    );
  }
}

export const CustomerDetailsWithAuthErrorHandler = withAuthErrorHandler(CustomerDetails);

export const CustomerDetailsWithIntl = injectIntl(CustomerDetails);

export const CustomerDetailsWithData = graphql(CustomerDetailsQuery, CustomerDetailsQueryOptions)(
  CustomerDetailsWithIntl,
);

export default withUserData(CustomerDetailsWithData);
