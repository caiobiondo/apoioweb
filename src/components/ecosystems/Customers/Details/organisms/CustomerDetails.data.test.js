import { CustomerDetailsQuery, CustomerDetailsQueryOptions } from './CustomerDetails.data';

describe('CustomerDetailsQuery', () => {
  it('queries correctly', () => {
    expect(CustomerDetailsQuery).toMatchSnapshot();
  });

  it('queries with defined options correctly', () => {
    // given
    const props = {
      match: {
        params: {
          customerId: 1,
        },
      },
    };

    // when
    const options = CustomerDetailsQueryOptions.options(props);

    // then
    expect(options).toEqual({
      cachePolicy: 'no-cache',
      forceFetch: true,
      variables: {
        customerId: 1,
      },
    });
  });
});
