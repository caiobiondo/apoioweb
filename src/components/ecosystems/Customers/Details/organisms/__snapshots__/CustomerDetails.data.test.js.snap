// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerDetailsQuery queries correctly 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "FetchCustomerDetails",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "customerId",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "customerId",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "customer",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "nickname",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "name",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "gender",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "comment",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "emails",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "email",
                        },
                        "selectionSet": null,
                      },
                    ],
                  },
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "phones",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "phone",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "provider",
                        },
                        "selectionSet": null,
                      },
                    ],
                  },
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "addresses",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "additional_address",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "city",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "neighborhood",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "street_name",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "street_number",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "state",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "zipcode",
                        },
                        "selectionSet": null,
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Int",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "customerId",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 413,
    "start": 0,
  },
}
`;
